### PR TITLE
Add error codes to rendered error templates

### PR DESCRIPTION
**Pull request by magickris93**
- Add error codes to rendering error templates.

### CODE REVIEW

File: dashboard/controller.py [hunk 1]
```diff
@@ -23,21 +23,21 @@
...
+    return render_template('error.html'), 500
 
 
 @error_handlers.app_errorhandler(PipelineNotFound)
 @error_handlers.app_errorhandler(PageOutOfRange)
 @error_handlers.app_errorhandler(404)
 def error_404(error):
     current_app.logger.error(f"{error}; raised on URL: {request.url}")
-    return render_template('error_404.html')
+    return render_template('error_404.html'), 404
```
Comment (kam193) [inline dashboard/controller.py]: I think, now we should check return code also in integration tests (e.g. when page should return 404, check if ``rv.status_code`` is 404).

File: dashboard/controller.py [hunk 2]
```diff
@@ -23,21 +23,21 @@
 @error_handlers.app_errorhandler(Exception)
 def generic_error(error):
     current_app.logger.error(f"{error}; raised on URL: {request.url}")
-    return render_template('error.html')
+    return render_template('error.html'), 500
```
Comment (pawelzny) [inline dashboard/controller.py]: I am almost 100% sure you want to make response with error code, not just return it as tuple element.
You can read more about it at: 

- http://flask.pocoo.org/docs/1.0/api/?highlight=make_response#flask.make_response
- http://docs.python-requests.org/en/master/api/?highlight=codes#status-code-lookup

```python
return make_response(render_template('error.html'), requests.codes.internal_server_error)
```

File: dashboard/controller.py (hunk 1 above)
Comment (pawelzny) [reply dashboard/controller.py]: Yes, we should check error codes but not as tuple value but as HTTP response status code.
```python
return make_response(render_template('error.html'), requests.codes.not_found)
```

File: dashboard/controller.py [hunk 3]
```diff
@@ -23,21 +23,21 @@
...
     current_app.logger.error(f"{error}; raised on URL: {request.url}")
-    return render_template('error_404.html')
+    return render_template('error_404.html'), 404
 
 
 @error_handlers.app_errorhandler(AuthenticationFailed)
 def auth_error(error):
     current_app.logger.error(f"{error}; raised on URL: {request.url}")
-    return render_template('auth_error.html')
+    return render_template('auth_error.html'), 401
```
Comment (pawelzny) [inline dashboard/controller.py]:
```python
return make_response(render_template('error.html'), requests.codes.unauthorized)
```

**Pull request merged**